tests/staticfiles_tests/test_storage.py

```diff
@@ -260,14 +260,14 @@ def test_css_import_case_insensitive(self):
 
     def test_js_source_map(self):
         relpath = self.hashed_file_path('cached/source_map.js')
-        self.assertEqual(relpath, 'cached/source_map.9371cbb02a26.js')
         with storage.staticfiles_storage.open(relpath) as relfile:
             content = relfile.read()
             self.assertNotIn(b'//# sourceMappingURL=source_map.js.map', content)
             self.assertIn(
                 b'//# sourceMappingURL=source_map.js.99914b932bd3.map',
                 content,
             )
+        self.assertEqual(relpath, 'cached/source_map.83947073cd1b.js')
```

Comment:
Move the assertion back before the `with ...` to keep the same pattern as the other tests?

Reply:
I moved it because this assertion fails first otherwise, and you can't see the behaviour under test, without fixing it first. 

I could update the other tests? 🤔

Happy to go either way really. (Once it's fixed the order does't matter, until next time.)

Reply:
I would move it back because this PR is intended for backporting. 

> I could update the other tests?

Agreed, we can update this and other tests in a separate cleanup.

Reply:
> I moved it because this assertion fails first otherwise, and you can't see the behaviour under test, without fixing it first.

Ah, fair enough.

> Agreed, we can update this and other tests in a separate cleanup.

👍